lib/super_id/models/class_methods.rb

```diff
@@ -91,19 +111,14 @@ def use_super_id_for(id_name_or_names, options={})
         super_id_names.each do |id_name|
           self.send(:define_method, "#{id_name.to_s}") do |*args|
             id = super(*args)
-
-            foreign_key_class = self.class.reflections.values.detect { |r| r.foreign_key == id_name.to_s }.try(:klass)
-
-            options = if foreign_key_class
-              foreign_key_class.super_id_options
-            else
-              self.class.super_id_options
-            end
-
-            self.class.make_super(id.to_i, options) if id
+            self.class.create_super_id(super(*args), id_name) if id
           end
         end
 
+        define_singleton_method('where') do |args|
```

Comment:
This does not account for code such as:

`Tag.where('banner_id = ? OR panel_id = ?', id, id)`